docs/blog/2019-12-12-how-we-accidentally-launched-a-popular-gatsby-plugin/index.md

```diff
@@ -0,0 +1,104 @@
+
+- [Gatsby’s recommended caching headers](https://www.gatsbyjs.org/docs/caching/) were not applied — not very good for your lighthouse scores, and one of the primary reasons we chose Gatsby in the first place.
+
+- Gatsby allows you to have [routes that only exist on the client side](/docs/building-apps-with-gatsby/#client-only-routes--user-authentication).
+
+- Essentially, you declare a starting point — say for example: /users/, and anything past that starting point will get picked up by the client side. Once the client navigates to /users/1, it will dynamically fetch that user from some sort of API. This is great and allows for very hybrid, partially static, partially dynamic applications. However, when people directly navigate to /users/1 , they will get a 404 because it simply does not exist on the serverside, which is kind of an issue.
+
+- We already had a CI service ourselves, and weren’t really interested in learning all these Amplify-specific things that we already had working ourselves just fine.
+
+- Being able to add additional metadata (such as different headers for certain files) was absent.
+
+So, I decided to experiment a bit and a week later, launched a plugin that fixed all of the afore mentioned problems:
+
+- Gatsby redirects now work just like they do on other hosting providers, and can be configured to be permanently or temporarily.
+
+- It applies the recommended caching headers by default which can also be fully customised.
+
+- If a client route is requested from the server side, it redirects them back to the starting point of the client route. (aka /users/1 now redirects to /users/ instead of 404'ing). Preferably we’d rewrite the url completely, but this is not possible with S3 and [gatsby-plugin-netlify](https://www.npmjs.com/package/gatsby-plugin-netlify) does exactly the same.
+
+- Unlike Amplify, it can be ran from your own infrastructure (or an EC2 instance — in which case it won’t need any configuration because it uses the AWS SDK which can automatically resolve the needed credentials!)
+
+- You can now set additional metadata on objects, meaning you can (for example) specify [a custom content type for your objects](https://github.com/jariz/gatsby-plugin-s3/blob/master/recipes/custom-content-type.md).
+
+You can get it here:
+[gatsbyjs.org/packages/gatsby-plugin-s3](/packages/gatsby-plugin-s3/?=plugin-s3)
+
+## The result
+
+We launched a beautiful, fast website that was cheap to host and didn’t even flinch whilst serving a million requests during our peak day. Outside of the festival period, costs were minimal. We completely proved that this stack is great for this type of project.
+
+[Check it out here.](https://www.oerol.nl/en/)
+
+Building the plugin allowed us to scale it as far as we needed it without a hitch, with the need for redirects and custom headers showing up fairly quickly as the project went on.
```

Comment:
Should this be "without the need for redirects"?

Reply:
I meant that as the project furthered, redirects became required for the project itself, which was not a problem, because plugin-s3 already had this.
Should I reword it? :)